index.d.ts

```diff
@@ -0,0 +1,109 @@
+import { WriteStream } from "fs"
+
+interface Coercable {
```

Comment:
I think the name of this overly emphasises coercion, can we maybe call it something like `Comparator` ?
e.g. a Comparator is an assertion that compares two values, strictly or coercively